homeassistant/components/sensor/ebusd.py

```diff
@@ -0,0 +1,185 @@
+
+
+def switch_format(value):
+    if value == 1:
+        return STATE_ON
+    return STATE_OFF
+
+
+class EbusdData(object):
+    """Get the latest data from Ebusd."""
+
+    def __init__(self, circuit, time):
+        """Initialize the data object."""
+        self._circuit = circuit
+        self._time = time
+        self.value = None
+    
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self, name):
+        """Call the Ebusd API to update the data."""
+        command = BASE_URL.format(self._circuit, name, self._time)
+        try:
+            _LOGGER.info("Receiving data from ebusdctl for %s: %s", name, command)
+            r = subprocess.check_output(command,shell=True, timeout=15)
+            command_result = r.strip().decode('utf-8')
+            if command_result == 'ERR: element not found':
+                _LOGGER.warning('ERR: element not found: %s', command)
+                self.value = STATE_UNKNOWN
+                return False
+            else:
+                self.value = command_result
+                return True
+        except subprocess.CalledProcessError:
+            _LOGGER.error("Command failed: %s", command)
+            return False
+        except subprocess.TimeoutExpired:
+            _LOGGER.error("Timeout for command: %s", command)
+            return False
+
+
+class Ebusd(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, data, sensor_type, name):
+        """Initialize the sensor."""
+        self._state = None
+        self._last_updated = None
+        self._client_name = name
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._icon = SENSOR_TYPES[sensor_type][2]
+        self._type = SENSOR_TYPES[sensor_type][3]
+        self.data = data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self._client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of this device."""
+        attr = {}
+        if self._last_updated is not None:
+            attr['Last Updated'] = self._last_updated
+        return attr
+
+    def update(self):
+        """Fetch new state data for the sensor.
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        update_result = False
+        if self._last_updated is None or (datetime.now() - self._last_updated) > timedelta(minutes=15):
+            update_result = self.data.update(self._name)
+
+        if update_result is True:
```

Comment:
Simple check if `self.data.value` is not `None` after storing it in a var and re-using that var later.